infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/SessionManager.java

```diff
@@ -46,18 +54,20 @@ public class SessionManager {
 
     private final CorfuRuntime runtime;
 
-    private String localCorfuEndpoint;
+    private final String localCorfuEndpoint;
 
     private CorfuReplicationManager replicationManager;
 
-    private LogReplicationConfigManager configManager;
+    private final LogReplicationConfigManager configManager;
 
     @Getter
-    private Set<LogReplicationSession> sessions = new HashSet<>();
+    private final Set<LogReplicationSession> sessions = new HashSet<>();
```

Comment:
I think the access to the methods in this class is single threaded because it is mostly done through FSM. If that is the case please update comments, else you need to figure out how to make it threadsafe.

Reply:
the access to the hashSets in this class are the only ones that is multi threaded.
They need to be protected from "remove" operation in the clientServerRouter.

In my last push, these structures are protected when accessed from the clientServerRouter. Protection in this class is not needed as the remove operation is single threaded, only the gets can be multi threaded in the clientServerRouter.

Also, in my last push, these have been changed to concurrent hash sets.

Reply:
Also, which comment indicates that the class may be accessed by multi threads? Not able to find it...